server/src/main/java/org/elasticsearch/action/index/IndexRequest.java

```diff
@@ -605,4 +608,15 @@ public IndexRequest setShardId(ShardId shardId) {
         throw new UnsupportedOperationException("shard id should never be set on IndexRequest");
     }
 
+    /**
+     * Override this method from {@link ReplicationRequest}. 6.4.0 introduces `is_write_index` to aliases so that
+     * index requests will resolve to the appropriate write index when available.
+     */
+    @Override
+    public IndicesOptions indicesOptions() {
+        if (Version.CURRENT.onOrAfter(Version.V_6_4_0)) {
+            return IndicesOptions.strictAliasToWriteIndexNoExpandForbidClosed();
+        }
+        return super.indicesOptions();
```

Comment:
Wy do we need this if? Seems like we could rather push this change to the proper branches without this conditional?